### DISCUSSION TITLE
Added: LDM Notes for August 23rd, 2021

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#5108] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-08-23.md

## Agenda

1. Nullability differences in partial type base clauses
2. Top-level statements default type accessibility
3. Lambda expression and method group type inference issues
    1. Better function member now ambiguous in some cases
    2. Conversions from method group to `object`
4. Interpolated string betterness in older language versions

**Comment by DavidArno** [comment]
The statement "*We also think that the likelihood is that most test projects are going to need IVT to their original at some point...*" is desperately disappointing. Anyone who uses IVT for tests needs to go back to testing school as they are doing it wrong. The idea that the language team are resigned to the fact that almost everyone will just do it wrong, rather than trying to encourage better testing behaviour, is very disappointing.

But at least the team reached what I see as the correct conclusion - to put top level statements inside an internal class - even if it is by an unfortunate route.

**Reply by jnm2** [reply]
Guess I need to go back to testing school.

**Reply by yaakov-h** [reply]
You, me, Microsoft...

**Reply by DavidArno** [reply]
Looks like it. You both might find this a useful starting point: https://blog.ploeh.dk/2015/09/22/unit-testing-internals/

**Comment by CyrusNajmabadi** [comment]
@DavidArno I've hidden your post due to the negative way it seems to be attacking people who participate in a very common pattern. If you can find a way to write it without the disparagement, I can make it visible again. 

Thanks and remember to try to keep things civil and professional here.

**Reply by DavidArno** [reply]
I appreciate you taking the time to explain your decision, but stand by my comment. So it'll have to remain hidden.

**Comment by huoyaoyuan** [comment]
IVT is especially useful where public api need to be audited, for example many first-party dotnet repositories.
Let's going to the topic for `Main`. In my view, top level statements is only useful for tiny applications with less than 3 files, which is very unlikely to have unit tests. *Changing top level statement to explicitly Main* isn't costful, so we don't need full feature parity.
